webapp/channels/src/actions/file_actions.ts

```diff
@@ -87,6 +87,20 @@ export function uploadFile({file, name, type, rootId, channelId, clientId, onPro
                     ]));
 
                     onSuccess(response, channelId, rootId);
+                } else if (xhr.readyState === 4 && xhr.status === 400) {
```

Comment:
I'm thinking we should cover other status codes just in case to cover any sort of error. Also suggest to follow the same order/pattern of conditional checking as above

```suggestion
                } else if (xhr.status >= 400 && xhr.readyState === 4) {
```